src/main/java/com/banking/app/bankingApp/controller/UserController.java

```diff
@@ -52,14 +57,21 @@ public UserController() {
 
     @GetMapping(value = "/users")
 
-    public ResponseEntity<List<User>> getAllUsers() {
-        return ResponseEntity.status(HttpStatus.OK).body(userManagementService.getAllUsers());
+    public ResponseEntity<List<User>> getAllUsers(@RequestHeader(name="Authorization")String authorization) {
+        try{
+            authorizationValidationRules.checkGetAllUsersRole(tokenUtil.getIdFromToken(authorization));
+            return ResponseEntity.status(HttpStatus.OK).body(userManagementService.getAllUsers());
+    }catch (RuntimeException e){
+        return ResponseEntity.status(HttpStatus.FORBIDDEN).build();
```

Comment:
Any runtime exception would result in 403 returned.
Throw custm exception in the role check and catch it here.
Runtime is too generic